### PR TITLE
EQ FIR: Enable testing with gcc builds and add check for filter length for graceful playback stop

### PR DESCRIPTION
**Pull request by singalsu**
This pull request addresses issue #2225. It reduces the tap count of one example FIR EQ configuration to enable testing of a more complex response than a flat/pass FIR response with gcc build.

There's add of check for a build specific filters length limitation that is now enforced for non-xcc xtensa builds.

A small code optimization is made to generic FIR core though the performance gain is quite low.

### CODE REVIEW

**Comment by singalsu** [comment]
Note: PR #2221 is also needed to fully fix the issue.

**Comment by lgirdwood** [comment]
CI usual suspects

**Comment by singalsu** [comment]
I added third commit for Kconfig update with note about restriction when used as xtensa gcc build.

File: src/audio/Kconfig [hunk 1]
```diff
@@ -24,7 +24,11 @@ config COMP_FIR
 	bool "FIR component"
 	default y
 	help
-	  Select for FIR component
+	  Select for FIR component. Note that FIR equalizer is recommended
+	  to be used in Xtensa DSP architecture when built with xt-xcc
+	  compiler only. As xtensa gcc build the max. filter tap count is
+	  severely restricted due to slow multiply-and-accumulate performance
+	  in generic C version.
```
Comment (lgirdwood) [inline src/audio/Kconfig]: I would re-write this to be more generic. i.e. "FIR performance can differ between DSP ISA and compiler used. FIR will perform best when MAC instructions are available in ISA and are used by the compiler. i.e. xt-xcc on xtensa will generate MAC instructions but GCC on xtensa wont. Filter tap cound can be severly restricted to reduce FIR cycles and FIR performance for DSP/compilers with no MAC support".

**Comment by lgirdwood** [comment]
@singalsu looks like CI is reporting an issue ?

File: src/audio/Kconfig (hunk 1 above)
Comment (singalsu) [reply src/audio/Kconfig]: Thanks, I'll replace. It's much better than my version.

**Comment by singalsu** [comment]
> @singalsu looks like CI is reporting an issue ?

I see only two kernel module fails for BSW_CYN_MAX98090 and "WARNING: quoted string split across lines" for patch check. The CI System/merge/build looks all green despite the red x in this page. I'll update PR and see if it was some CI glitch.

File: src/audio/Kconfig (hunk 1 above)
Comment (singalsu) [reply src/audio/Kconfig]: Done!

**Pull request merged**